pkg/webhook/instance_admission.go

```diff
@@ -193,6 +185,19 @@ func admitUpdate(old, new *kudov1beta1.Instance, ov *kudov1beta1.OperatorVersion
 	isDeleting := new.IsDeleting() // a non-empty meta.deletionTimestamp is a signal to switch to the uninstalling life-cycle phase
 	isPlanTerminal := isTerminal(new, newPlan, new.Spec.PlanExecution.UID)
 
+	paramDiff := kudov1beta1.ParameterDiff(old.Spec.Parameters, new.Spec.Parameters)
+	paramDefs, err := kudov1beta1.GetParamDefinitions(paramDiff, ov)
+	if err != nil && !isUpgrade { // we allow removing parameters when upgrading to a new OV
```

Comment:
I understand why we ignore removed parameters when upgrading, but what about an added parameter that isn't in the new OV? How about returning two diffs in `ParameterDiff`, one for added parameters, another for removed parameters. This would allow us to distinguish between these different cases.

Reply:
There's a check in `triggeredByParameterUpdate` if the parameter used is in the OV. So this case is handled there. Feel free to ignore, though it would be great to already handle this here.

Reply:
You mean in [L307](https://github.com/kudobuilder/kudo/pull/1441/files#diff-c10c73f4220d3a0c71475b7002482b65R307)? This is for when the _plan_ does not exist in OV, not the _parameter_ itself.

Reply:
> I understand why we ignore removed parameters when upgrading, but what about an added parameter that isn't in the new OV?

That's correct and I thought about returning two diffs but wasn't sure if the problem is worth the complexity. Wdyt?

Reply:
Oh, it's for a plan, didn't see this. IMO, it's worth the complexity, as it still concerns invalid Instance parameters during updates.

Reply:
Convinced and... fixed!